src/sources/datadog_agent/mod.rs

```diff
@@ -274,15 +274,22 @@ impl SourceConfig for DatadogAgentConfig {
             )
             .with_standard_vector_source_metadata();
 
+        let mut output = Vec::new();
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏this is kind of a mega nit since I don't think this function is evoke outside of startup but I believe we expect the resulting vector returned to be at least length 1.

```suggestion
        let mut output = Vec::with_capacity(1);
```